packages/store/addon/-private/system/core-store.ts

```diff
@@ -2904,11 +2988,49 @@ class Store extends Service {
     return internalModelFactoryFor(this).lookup(modelName, id, lid);
   }
 
-  _createRecordData(modelName, id, clientId): RecordData {
+  serializeRecord(record: Record, options?: Dict<string, unknown>): unknown {
+    if (CUSTOM_MODEL_CLASS) {
+      let identifier = recordIdentifierFor(record);
+      let internalModel = internalModelFactoryFor(this).peek(identifier.type, identifier.id, identifier.lid);
+      // TODO we used to check if the record was destroyed here
+      return internalModel!.createSnapshot(options).serialize(options);
```

Comment:
```suggestion
      // because a `Record` is provided there will always be a matching internalModel
      return internalModel!.createSnapshot(options).serialize(options);
```

alt: use lookup